iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -155,29 +163,34 @@ cxx::expected<bool, IpcChannelError> UnixDomainSocket::unlinkIfExists(const NoPa
     }
 }
 
-cxx::expected<IpcChannelError> UnixDomainSocket::destroy() noexcept
+cxx::expected<IpcChannelError> UnixDomainSocket::closeFd(int32_t fileDescriptor)
 {
-    if (m_sockfd != INVALID_FD)
-    {
-        auto closeCall = cxx::makeSmartC(
-            closePlatformFileHandle, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {ERROR_CODE}, {}, m_sockfd);
+    auto closeCall = cxx::makeSmartC(
+        closePlatformFileHandle, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {ERROR_CODE}, {}, fileDescriptor);
 
-        if (!closeCall.hasErrors())
+    if (!closeCall.hasErrors())
+    {
+        if (IpcChannelSide::SERVER == m_channelSide)
         {
-            if (IpcChannelSide::SERVER == m_channelSide)
-            {
-                unlink(m_sockAddr.sun_path);
-            }
+            unlink(m_sockAddr.sun_path);
+        }
 
-            m_sockfd = INVALID_FD;
-            m_isInitialized = false;
+        m_sockfd = INVALID_FD;
+        m_isInitialized = false;
 
-            return cxx::success<void>();
-        }
-        else
-        {
-            return createErrorFromErrnum(closeCall.getErrNum());
-        }
+        return cxx::success<void>();
+    }
+    else
+    {
+        return createErrorFromErrnum(closeCall.getErrNum());
+    }
+}
+
+cxx::expected<IpcChannelError> UnixDomainSocket::destroy() noexcept
+{
+    if (m_isInitialized)
+    {
+        return closeFd(m_sockfd);
```

Comment:
Nope, inside `UnixDomainSocket::createSocket()`,  `closeFd()` is called with a different value. Using a default parameter instead as an alternative, is less readable for me, because I have to be aware of a default param.

Reply:
Changed return value of `createSocket()` and rename it to `initalizeSocket()`